documentation/src/main/asciidoc/migration/index.asciidoc

```diff
@@ -389,4 +389,9 @@ query that returns the same results.
 Possible changes can include: some clauses can be rearranged, nested bool queries might be flattened, a bool query might be
 replaced with its clause.
 
-A bool query with a single `mustNot` clause and applied boost would implicitly add a `must` with `match_all` clause.
\ No newline at end of file
+A bool query with a single `mustNot` clause and applied boost would implicitly add a `must` with `match_all` clause.
+
+Any of `id`/`entity`/`entityReference`/`documentReference`/`score` as well as Lucene-specific `document`/`explanation`
+and Elasticsearch-specific `jsonHit`/`source`/`explanation` projections are now not allowed within a context of a nested projection (`object`).
+These projections are targeting the root object and placing them inside a nested projection might lead to confusing results.
+Using these within such `object` projection will now cause an exception.
```

Comment:
> let users know in the migration guide about the change

This is not necessary as the `object` projection was introduced in 6.2.0.Alpha. We don't document breaking changes between alphas in the migration guide.